### PR TITLE
Remove duplicates in setup.py and run rudimentary verifications

### LINKED ISSUE #2009: Feast CLI error - "No module named 'proto'"
## Expected Behavior 
Create a feature repo as expected.

## Current Behavior
Feast dies with an error when running the following:

~ ❯ feast init feature_repo                                                                                    
Traceback (most recent call last):
  File "/home/richard/anaconda3/envs/feast/bin/feast", line 5, in <module>
    from feast.cli import cli
  File "/home/richard/anaconda3/envs/feast/lib/python3.7/site-packages/feast/__init__.py", line 12, in <module>
    from .feature_service import FeatureService
  File "/home/richard/anaconda3/envs/feast/lib/python3.7/site-packages/feast/feature_service.py", line 6, in <module>
    from feast.base_feature_view import BaseFeatureView
  File "/home/richard/anaconda3/envs/feast/lib/python3.7/site-packages/feast/base_feature_view.py", line 19, in <module>
    from proto import Message
ModuleNotFoundError: No module named 'proto'

## Environment
Python 3.7.11
Created a brand new conda environment and then ran `pip install feast`, which installed the following:
```
Successfully installed Click-7.1.2 PyYAML-6.0 anyio-3.3.4 asgiref-3.4.1 attrs-21.2.0 cachetools-4.2.4 charset-normalizer-2.0.7 colorama-0.4.4 dill-0.3.4 fastapi-0.70.0 fastavro-1.4.7 feast-0.15.0 google-api-core-2.2.2 google-auth-2.3.3 googleapis-common-protos-1.52.0 grpcio-1.42.0rc1 grpcio-reflection-1.42.0rc1 h11-0.12.0 httptools-0.2.0 idna-3.3 importlib-metadata-4.8.2 importlib-resources-5.4.0 jsonschema-4.2.1 mmh3-3.0.0 numpy-1.21.4 pandas-1.3.4 pandavro-1.5.2 protobuf-3.19.1 pyarrow-6.0.0 pyasn1-0.4.8 pyasn1-modules-0.2.8 pydantic-1.8.2 pyrsistent-0.18.0 python-dateutil-2.8.2 python-dotenv-0.19.1 pytz-2021.3 requests-2.26.0 rsa-4.7.2 six-1.16.0 sniffio-1.2.0 starlette-0.16.0 tabulate-0.8.9 tenacity-8.0.1 toml-0.10.2 tqdm-4.62.3 typing-extensions-3.10.0.2 urllib3-1.26.7 uvicorn-0.15.0 uvloop-0.16.0 watchgod-0.7 websockets-10.0 zipp-3.6.0
```

No other packages installed beforehand.

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:

This removes some of the duplicates of packages in the setup.py, to make sure we use consistent versions.

Additionally, when packages are not compatible we should an error like this:

```
$ pip-compile --dry-run sdk/python/setup.py --extra dev
ERROR: ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
ERROR: grpcio-reflection 1.42.0rc1 requires grpcio>=1.42.0rc1, but you have grpcio 1.41.1 which is incompatible.
Could not find a version that matches grpcio<1.40,<2.0dev,>=1.34.0,>=1.38.1,>=1.41.1 (from feast (sdk/python/setup.py))
Tried: 0.13.0, 0.13.1, 0.14.0, 0.15.0, 1.0.0, 1.0.1, 1.0.2, 1.0.3, 1.0.4, 1.1.0, 1.1.3, 1.2.0, 1.2.1, 1.3.0, 1.3.5, 1.4.0, 1.6.0, 1.6.3, 1.7.0, 1.7.3, 1.8.1, 1.8.2, 1.8.3, 1.8.4, 1.8.6, 1.9.0, 1.9.1, 1.10.0, 1.10.1, 1.11.0, 1.11.1, 1.12.0, 1.12.1, 1.13.0, 1.13.0, 1.14.0, 1.14.0, 1.14.1, 1.14.1, 1.14.2, 1.14.2, 1.15.0, 1.15.0, 1.16.0, 1.16.0, 1.16.1, 1.16.1, 1.17.0, 1.17.0, 1.17.1, 1.17.1, 1.18.0, 1.18.0, 1.19.0, 1.19.0, 1.20.0, 1.20.0, 1.20.1, 1.20.1, 1.21.1, 1.21.1, 1.22.0, 1.22.0, 1.22.1, 1.22.1, 1.23.0, 1.23.0, 1.23.1, 1.23.1, 1.24.0, 1.24.0, 1.24.1, 1.24.1, 1.24.3, 1.24.3, 1.25.0, 1.25.0, 1.26.0, 1.26.0, 1.27.1, 1.27.1, 1.27.2, 1.27.2, 1.28.1, 1.28.1, 1.29.0, 1.29.0, 1.30.0, 1.30.0, 1.31.0, 1.31.0, 1.32.0, 1.32.0, 1.33.1, 1.33.1, 1.33.2, 1.33.2, 1.34.0, 1.34.0, 1.34.1, 1.34.1, 1.35.0, 1.35.0, 1.36.0, 1.36.0, 1.36.1, 1.36.1, 1.37.0, 1.37.0, 1.37.1, 1.37.1, 1.38.0, 1.38.0, 1.38.1, 1.38.1, 1.39.0, 1.39.0, 1.40.0, 1.40.0, 1.41.0, 1.41.0, 1.41.1, 1.41.1
Skipped pre-versions: 0.4.0a0, 0.4.0a1, 0.4.0a2, 0.4.0a3, 0.4.0a4, 0.4.0a5, 0.4.0a6, 0.4.0a7, 0.4.0a8, 0.4.0a13, 0.4.0a14, 0.5.0a0, 0.5.0a1, 0.5.0a2, 0.9.0a0, 0.9.0a1, 0.10.0a0, 0.11.0b0, 0.11.0b1, 0.12.0b0, 0.13.1rc1, 0.14.0rc1, 1.0.0rc1, 1.0.0rc2, 1.0.1rc1, 1.9.0rc1, 1.9.0rc2, 1.9.0rc3, 1.10.0rc1, 1.10.0rc2, 1.10.1rc1, 1.10.1rc2, 1.11.0rc1, 1.11.0rc2, 1.11.1rc1, 1.12.0rc1, 1.13.0rc1, 1.13.0rc2, 1.13.0rc3, 1.14.0rc1, 1.14.0rc1, 1.14.0rc2, 1.14.0rc2, 1.14.2rc1, 1.14.2rc1, 1.15.0rc1, 1.15.0rc1, 1.16.0rc1, 1.16.0rc1, 1.16.1rc1, 1.16.1rc1, 1.17.0rc1, 1.17.0rc1, 1.17.1rc1, 1.17.1rc1, 1.18.0rc1, 1.18.0rc1, 1.19.0rc1, 1.19.0rc1, 1.20.0rc1, 1.20.0rc1, 1.20.0rc2, 1.20.0rc2, 1.20.0rc3, 1.20.0rc3, 1.21.0rc1, 1.21.0rc1, 1.21.1rc1, 1.21.1rc1, 1.22.0rc1, 1.22.0rc1, 1.23.0rc1, 1.23.0rc1, 1.24.0rc1, 1.24.0rc1, 1.25.0rc1, 1.25.0rc1, 1.26.0rc1, 1.26.0rc1, 1.27.0rc1, 1.27.0rc1, 1.27.0rc2, 1.27.0rc2, 1.28.0.dev0, 1.28.0.dev0, 1.28.0rc1, 1.28.0rc1, 1.28.0rc2, 1.28.0rc2, 1.28.0rc3, 1.28.0rc3, 1.30.0rc1, 1.30.0rc1, 1.31.0rc1, 1.31.0rc1, 1.31.0rc2, 1.31.0rc2, 1.32.0rc1, 1.32.0rc1, 1.33.0rc1, 1.33.0rc1, 1.33.0rc2, 1.33.0rc2, 1.34.0rc1, 1.34.0rc1, 1.35.0rc1, 1.35.0rc1, 1.36.0rc1, 1.36.0rc1, 1.37.0rc1, 1.37.0rc1, 1.38.0rc1, 1.38.0rc1, 1.39.0rc1, 1.39.0rc1, 1.40.0rc1, 1.41.0rc2, 1.41.0rc2, 1.42.0rc1, 1.42.0rc1
There are incompatible versions in the resolved dependencies:
  grpcio<1.40,>=1.34.0 (from feast (sdk/python/setup.py))
  grpcio<2.0dev,>=1.38.1 (from google-cloud-bigquery==2.30.1->feast (sdk/python/setup.py))
  grpcio>=1.34.0 (from grpcio-testing==1.34.0->feast (sdk/python/setup.py))
  grpcio>=1.41.1 (from grpcio-reflection==1.41.1->feast (sdk/python/setup.py))
  grpcio>=1.34.0 (from grpcio-tools==1.34.0->feast (sdk/python/setup.py))
[10/11/21 2:55:39] (python-3.7.12) ~/tecton/feast   achal/simplify-setup-py ● 
$ echo $?

2
```

**Which issue(s) this PR fixes**:

Fixes #2009 

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

File: sdk/python/setup.py [hunk 1]
```diff
@@ -215,7 +210,7 @@ def run(self):
     # https://stackoverflow.com/questions/28509965/setuptools-development-requirements
     # Install dev requirements with: pip install -e .[dev]
     extras_require={
-        "dev": ["mypy-protobuf==1.*", "grpcio-testing==1.*"],
+        "dev": DEV_REQUIRED,
```
Comment (felixwang9817) [inline sdk/python/setup.py:213]: nit: could you add `proto-plus` to `REQUIRED`? several folks have had [issues](https://github.com/feast-dev/feast/issues/2009) with this recently

Comment (achals) [reply sdk/python/setup.py:214]: Done

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Pull request merged**